src/clj/witan/send/model/input.clj

```diff
@@ -47,6 +47,24 @@
       (apply-schema-coercion schema)
       (as-> {:keys [column-names columns]} (ds/dataset column-names columns))))
 
+(defn is-dataset-erroneous? [dataset]
+  (->> dataset second :column-names first ((complement instance?) clojure.lang.Keyword)))
+
+(defn check-dataset [pred s datasets]
+  (when (some true? (map pred datasets))
```

Comment:
Do you need this given you filter on the predicate anyway?

Reply:
Yep you're right, not needed now it's a single pred